### PR TITLE
feature: add vesting token donation feature

### PR DESCRIPTION
**Pull request by YunSuk-Yeo**
## Description

Implement new interface to donate existing vesting tokens.

`MsgDonateAllVestingToken` can be executed when there is no delegated vesting token in the from account.

### New Codec
In order to enable tx execution with Ledger, the codecs for new message types are added.

`cosmos-sdk/MsgCreateVestingAccount`
`cosmos-sdk/MsgCreatePeriodicVestingAccount`
`cosmos-sdk/MsgDonateAllVestingTokens`

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/auth/vesting/msg_server.go [hunk 1]
```diff
@@ -158,3 +160,57 @@ func (s msgServer) CreatePeriodicVestingAccount(goCtx context.Context, msg *type
...
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "account %s has delegated vesting tokens", msg.FromAddress)
+	}
+
+	vestingCoins := vestingAcc.GetVestingCoins(ctx.BlockTime())
+	if vestingCoins.IsZero() {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "account %s has no vesting tokens", msg.FromAddress)
+	}
+
+	// Change the account as normal account
+	ak.SetAccount(ctx,
```
Comment (JSHan94) [inline x/auth/vesting/msg_server.go:195]: Is it okay to change the vesting account to  base account before FundCommunityPool?

File: x/auth/vesting/msg_server.go [hunk 2]
```diff
@@ -158,3 +160,57 @@ func (s msgServer) CreatePeriodicVestingAccount(goCtx context.Context, msg *type
...
+	}
+
+	acc := ak.GetAccount(ctx, from)
+	if acc == nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "account %s not exists", msg.FromAddress)
+	}
+
+	vestingAcc, ok := acc.(exported.VestingAccount)
+	if !ok {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "account %s not vesting account", msg.FromAddress)
```
Comment (JSHan94) [inline x/auth/vesting/msg_server.go:182]: `"account %s is not vesting account"`

File: x/auth/vesting/msg_server.go (hunk 1 above)
Comment (YunSuk-Yeo) [reply x/auth/vesting/msg_server.go:195]: We can fund the community pool after we change the vesting account to the base account.
If not, we cannot send the fund due to vesting.

Comment (JSHan94) [reply x/auth/vesting/msg_server.go:195]: I see, got it

**Comment by JSHan94** [comment]
LGTM

**Comment by JSHan94** [comment]
LGTM

**Review by JSHan94 — APPROVED** [review]

**Pull request merged**